tests/mocks/services.py

```diff
@@ -1184,6 +1188,7 @@ def __init__(
         items: List[RegistryItem],
         package_type: PackageType.ValueType,
     ):
+        self.page_token = None
```

Comment:
per above comment, this should be deleted

Reply:
I'm not sure I'm understanding "such that they don't exist in general", but I ended up keeping them but setting to default values because when GetFragmentHistory tried to access them, the class attributes did not exist which gave me errors. Then in GetFragmentHistory, they were set based on the values in the request - it seems like this addressed your concern of setting twice, but if there's a more appropriate way to do this let me know!

Reply:
Curious what you mean by "tried to access them"? You should be able to just set to them (i.e., `self.some_fake_value = "this is a fake value that wasn't defined prior!"` should work with no issues), but if you tried to access the value with `foo = self.some_other_fake_value_that_has_not_been_defined` then yeah, I'd expect that to fail.